charmcraft/providers.py

```diff
@@ -105,6 +106,37 @@ def get_command_environment() -> Dict[str, str]:
     return env
 
 
+def clean_project_environments(
+    charm_name: str,
+    *,
+    lxd_project: str = "charmcraft",
+    lxd_remote: str = "local",
+) -> List[str]:
+    """Clean up any environments created for project.
+
+    :param charm_name: Name of project.
+    :param lxd_project: Name of LXD project.
+    :param lxd_remote: Name of LXD remote.
+
+    :returns: List of containers deleted.
+    """
+    deleted: List[str] = []
+    lxc = lxd.LXC()
+
+    for name in lxc.list_names(project=lxd_project, remote=lxd_remote):
+        match_regex = f"^charmcraft-{charm_name}-.*-.*-.*$"
```

Comment:
I wonder if the `.*` between dashes wouldn't be better matched using `[^-]+` (unless `charmcraft-name---` is a valid name).

Reply:
Fair point, I updated it to `r"^charmcraft-{charm_name}-.+-.+.+*$" ` which should address this case. Thanks!